tests/functional/base_test.go

```diff
@@ -94,37 +96,63 @@ func CreateOVNDBCluster(namespace string, OVNDBClusterName string, spec map[stri
 	return th.CreateUnstructured(raw)
 }
 
+func UpdateOVNDBCluster(cluster *ovnv1.OVNDBCluster) {
+	k8sClient.Update(ctx, cluster)
+}
+
 func OVNDBClusterConditionGetter(name types.NamespacedName) condition.Conditions {
 	instance := GetOVNDBCluster(name)
 	return instance.Status.Conditions
 }
 
+func ScaleDBCluster(name types.NamespacedName, replicas int32) {
+	Eventually(func(g Gomega) {
+		c := GetOVNDBCluster(name)
+		*c.Spec.Replicas = replicas
+		g.Expect(k8sClient.Update(ctx, c)).Should(Succeed())
+	}).Should(Succeed())
+}
+
 // CreateOVNDBClusters Creates NB and SB OVNDBClusters
-func CreateOVNDBClusters(namespace string, nad string) []types.NamespacedName {
+func CreateOVNDBClusters(namespace string, nad map[string][]string, replicas int32) []types.NamespacedName {
 	dbs := []types.NamespacedName{}
 	for _, db := range []string{v1beta1.NBDBType, v1beta1.SBDBType} {
 		name := fmt.Sprintf("ovn-%s", uuid.New().String())
 		spec := GetDefaultOVNDBClusterSpec()
+		string_nad := ""
+		for k, _ := range nad {
```

Comment:
No, I don't handle nad with no namespace, as IIRC if namespace is not passed the test will fail since at some point (during reconcile loop) will try to find that NAD but it won't be able to. (It's been a while since I handled this problem and I don't remember exactly the internals, will try to deep down later).
Regarding getting multiple NADs, it's not handled (Will add comment) since I dbcluster is not able to have multiples NADs ([type(spec.NetworkAttachment) == string](https://github.com/openstack-k8s-operators/ovn-operator/blob/main/config/crd/bases/ovn.openstack.org_ovndbclusters.yaml#L85C19-L89)).

Reply:
ok please fail / assert on invalid input (map of size > 1) so that the author of the next test case that may use the function is aware what happens.